src/app/api/notebooks/[id]/sources/route.ts

```diff
@@ -12,6 +12,16 @@ export async function POST(
       return new NextResponse("Unauthorized", { status: 401 });
     }
 
+    // Get the notebook to determine the provider
+    const notebook = await prisma.notebook.findUnique({
```

Comment:
Should pass in the notebook id as part of the route call instead of doing a database lookup.